### ISSUE TITLE
Add a readme section to the libraries to include azure-sdk-bom dependency as the first option

### ISSUE THREAD

**Issue by pallavit** [Azure/azure-sdk-for-java#22419]
We should add instructions in library readme's to first consider using BOMs and then add the direct dependency if needed. This will promote the use of BOM and prevent dependency conflicts as well.

- [ ] Update the SDK template to add BOM dependency.
- [ ] Update all SDKs to include BOM in the readmes.
- [ ] Update the template generator to also include the BOM in the readmes

**Comment by pallavit** [comment]
I was thinking about this a little more and ideally we would only want to add this section to the libraries that have GA'ed (as they are the ones in the BOM). Let me know if anyone disagrees.

Also, adding this section to the readme of a beta version of the library is like telling the customer - you landed on this version but consider using the BOM version instead. Could this be a source of confusion\frustration or is this a norm in general?

Here is a sample readme section, feedback\suggestion please!

> ### Include the Package
> #### Include the BOM file
> 
> Please include the azure-sdk-bom to your project to take dependency on the latest GA version of the library. Let me know what you think.
> 
> ```xml
> <dependencyManagement>
>     <dependencies>
>         <dependency>
>             <groupId>com.azure</groupId>
>             <artifactId>azure-sdk-bom</artifactId>
>             <version>1.0.4-beta.1</version>
>             <type>pom</type>
>             <scope>import</scope>
>         </dependency>
>     </dependencies>
> </dependencyManagement>
> ```
> 
> #### Include direct dependency
> If you want to take dependency on this particular version of the library, add the direct dependency to your project as follows.
> 
> [//]: # ({x-version-update-start;com.azure:azure-ai-anomalydetector;current})
> ```xml
> <dependency>
>   <groupId>com.azure</groupId>
>   <artifactId>azure-ai-anomalydetector</artifactId>
>   <version>3.0.0-beta.1</version>
> </dependency>
> ```
> [//]: # ({x-version-update-end})

**Comment by srnagar** [comment]
I agree that `Include the BOM` should be for GA libraries only since preview libraries are not part of the BOM. In general, if a library appears in the BOM, update the README as shown above. 

Also, just a minor fix to the above template:

```xml
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>com.azure</groupId>
            <artifactId>azure-sdk-bom</artifactId>
            <version>1.0.4-beta.1</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
    </dependencies>
</dependencyManagement>
...

<dependencies>
      ....
      <dependency>
           <groupId>com.azure</groupId>
           <artifactId>azure-ai-anomalydetector</artifactId>
      </dependency>
       ....
</dependencies>
```

The bom should be included in the `dependencyManagement` section and the artifact should be included in the `dependencies` section.

**Comment by JonathanGiles** [comment]
I'm curious what your thoughts are on this: if we give different guidance to beta / non-beta libraries, because non-beta API is not in the BOM, are we risking people upgrading from beta to non-beta versions of a dependency without taking the time to introduce the BOM, and therefore reducing the set of BOM users? Would we be better to include beta versions in a BOM so that our guidance can be the same for all libraries?

**Comment by JonathanGiles** [comment]
I would also be inclined to link out from the readme text about the BOM to a docs.microsoft link that details more about the BOM, why it exists, how it helps, problem solving, etc.

**Comment by pallavit** [comment]
> I would also be inclined to link out from the readme text about the BOM to a docs.microsoft link that details more about the BOM, why it exists, how it helps, problem solving, etc.

I could not find any documentation on this, is this something we have to write? If so, I will create a work-item for this.

> I'm curious what your thoughts are on this: if we give different guidance to beta / non-beta libraries, because non-beta API is not in the BOM, are we risking people upgrading from beta to non-beta versions of a dependency without taking the time to introduce the BOM, and therefore reducing the set of BOM users? Would we be better to include beta versions in a BOM so that our guidance can be the same for all libraries?

That is a good question! We are definitely risking people not moving. However, I do not think we should include beta libraries in BOM just because we already have beta libraries that have not GA'ed in a long time and may never GA, Including their dependencies in the BOM, maintaining them for BOMs feels like an overkill. We can choose to include a section in readme that links to the said BOM documentation and talks about it and then may be say ... once the library GA's consider moving to BOM instead - people will be informed of it, but they may choose to not use the BOM. 


Are there any metrics on how many people take dependency on beta libraries, before their first GA? Also, your first question makes me ask - have we seen dependency issues\customer concerns for beta libraries? If so, do we have any idea on how many of those are for the libraries that have already GA'ed so have adoption vs libraries that have never GA'ed.

**Comment by srnagar** [comment]
One issue with including beta libraries in BOM is that a beta library can depend on another beta version of a library that also has a GA version. For e.g. beta version of Web PubSub can take a dependency on a beta version of azure-core. Now, we can only have one version of azure-core in BOM and we want to generally include the GA version only.

**Comment by pallavit** [comment]
That is a good point!. In the same vein - they can have breaking changes, so customer's adding those dependencies explicitly might be cause less grief overall.

**Comment by JonathanGiles** [comment]
Please do create a work item on documenting the BOM. Happy to work with you on this and getting it up onto docs.microsoft.com.

**Comment by JonathanGiles** [comment]
Some libraries don't seem to be updated, e.g. storage?

**Comment by pallavit** [comment]
> Some libraries don't seem to be updated, e.g. storage?

I have an active PR out for storage - https://github.com/Azure/azure-sdk-for-java/pull/22858. We are working out the logistics with them but that is why the only reason the PR is not yet merged.
